src/pipeline/pipelineExplorer.ts

```diff
@@ -12,18 +12,27 @@ import * as path from 'path';
 const kubeConfigFolder: string = path.join(Platform.getUserHomePath(), '.kube');
 
 export class PipelineExplorer implements TreeDataProvider<TektonNode>, Disposable {
-  private static tkn: Tkn = TknImpl.Instance;
+  private tkn: Tkn;
+  private static instance: PipelineExplorer
   private treeView: TreeView<TektonNode>;
   private fsw: FileContentChangeNotifier;
   private onDidChangeTreeDataEmitter: EventEmitter<TektonNode | undefined> = new EventEmitter<TektonNode | undefined>();
   readonly onDidChangeTreeData: Event<TektonNode | undefined> = this.onDidChangeTreeDataEmitter.event;
 
   constructor() {
+    this.tkn = TknImpl.Instance;
     this.fsw = WatchUtil.watchFileForContextChange(kubeConfigFolder, 'config');
     this.fsw.emitter.on('file-changed', this.refresh.bind(this));
     this.treeView = window.createTreeView('tektonPipelineExplorerView', { treeDataProvider: this, canSelectMany: true });
   }
 
+  static getInstance(): PipelineExplorer {
+    if (!PipelineExplorer.instance) {
+      PipelineExplorer.instance = new PipelineExplorer();
+    }
+    return PipelineExplorer.instance;
+  }
+
```

Comment:
Is there any reason to use Java singletons in TypeScript? 
This module has `pipelineExplorer` const which you can use, it is more preferable way to get singleton pattern in TS/JS